spec/services/dqt_api_spec.rb

```diff
@@ -84,4 +84,87 @@
       end
     end
   end
+
+  describe ".unlock_teacher!" do
+    before do
+      FeatureFlag.activate(:unlock_teachers_self_service_portal_account)
+    end
+    after do
+      FeatureFlag.deactivate(:unlock_teachers_self_service_portal_account)
+    end
+
+    subject(:unlock_teacher!) do
+      described_class.unlock_teacher!(teacher_account)
+    end
+    let(:teacher_account) do
+      {
+        trn: "2921020",
+        emailAddresses: ["anonymous@anonymousdomain.org.net.co.uk"],
+        firstName: "Kevin",
+        lastName: "Evans",
+        dateOfBirth: "1990-01-01",
+        nationalInsuranceNumber: "AA123456A",
+        uid: "f7891223-7661-e411-8047-005056822391"
+      }
+    end
+
+    context "when teacher ID is found", vcr: true do
+      it { is_expected.to be_nil }
+    end
+
+    context "when teacher ID is not found", vcr: true do
+      let(:teacher_account) do
+        {
+          trn: "2921020",
+          emailAddresses: ["anonymous@anonymousdomain.org.net.co.uk"],
+          firstName: "Kevin",
+          lastName: "Evans",
+          dateOfBirth: "1990-01-01",
+          nationalInsuranceNumber: "AA123456A",
+          uid: "f6891223-7661-e431-8047-005056822391"
+        }
+      end
+      it { is_expected.to be_nil }
+    end
+
+    context "when teacher ID is not valid", vcr: true do
+      let(:teacher_account) do
+        {
+          trn: "2921020",
+          emailAddresses: ["anonymous@anonymousdomain.org.net.co.uk"],
+          firstName: "Kevin",
+          lastName: "Evans",
+          dateOfBirth: "1990-01-01",
+          nationalInsuranceNumber: "AA123456A",
+          uid: "f689122-7661-e431-8047-00506822391"
+        }
+      end
+      it { is_expected.to be_nil }
+    end
+
+    context "when uid key is not present" do
+      let(:teacher_account) do
+        {
+          trn: "2921020",
+          emailAddresses: ["anonymous@anonymousdomain.org.net.co.uk"],
+          firstName: "Kevin",
+          lastName: "Evans",
+          dateOfBirth: "1990-01-01",
+          nationalInsuranceNumber: "AA123456A"
+        }
+      end
+      it { is_expected.to be_nil }
```

Comment:
See above, here we can make `uid: nil`

Reply:
@tvararu we want to verify a KeyError is raised if the expected key is not present in the returned data. Setting this to nil will only pass an empty value but what we want is for the KeyError to be raised. Correct me if i am wrong